### ISSUE TITLE
promtool command for dynamic retention

### ISSUE THREAD

**Issue by gouthamve** [prometheus/prometheus#3628]
This would be a simple tool that hits the right endpoints based on a simple config-file that looks like this:

```yaml
# The frequency at which tombstone cleanup needs to be run.
# This means the /clean_tombstones endpoint is hit every 1h
cleanup_interval: 1h
# The frequency at which the deletes need to be run.
# This means that the /delete_series endpoint is hit every 5mins for the below deletes.
delete_interval: 5m

deletes:
  - queries:
      - {__name__=~"job1_.*"}
      - {job="tempJob"}
    retention: 10h # The retention time of every series that matches the above queries.
```

@brian-brazil Are you ok with including this in promtool?

**Comment by brian-brazil** [comment]
I'm not a fan of having anything in promtool require a config file (bar
validation/unittests), I think it should work entirely via the command line.
Nor do I think that promtool should be a daemon, as that's not very tool
like.

In relation to the particular feature I'd suggest having users use Cron for
a good while so we can see how this works out in terms of operations and
support rather than adding it to the core immediately.

On 27 Dec 2017 11:15, "Goutham Veeramachaneni" <notifications@github.com>
wrote:

This would be a simple tool that hits the right endpoints based on a simple
config-file that looks like this:

# The frequency at which tombstone cleanup needs to be run.# This
means the /clean_tombstones endpoint is hit every 1hcleanup_interval:
1h# The frequency at which the deletes need to be run.# This means
that the /delete_series endpoint is hit every 5mins for the below
deletes.delete_interval: 5m
deletes:
  - queries:
      - {__name__=~"job1_.*"}
      - {job="tempJob"}
    retention: 10h # The retention time of every series that matches
the above queries.

@brian-brazil <https://github.com/brian-brazil> Are you ok with including
this in promtool?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/prometheus/prometheus/issues/3628>, or mute the thread

**Comment by roidelapluie** [comment]
Can be closed because superseded by #1381 and https://groups.google.com/d/msgid/prometheus-developers/CANVFovWOAsu2doRykJfvZ--R5iVX9qHLAEtFfwa25p7b15Sg2w%40mail.gmail.com
